### PR TITLE
Move StitchedSurfaceMeshGenerator from BISON to MOOSE and rename

### LINKED ISSUE #26641: Move SitchedSurfaceMeshGenerator from BISON to MOOSE
## Reason

The existing mesh generator is generic and can be put into MOOSE for use.

## Design

Move mesh generator and rename to the a more accurate name: `StitchedBoundaryMeshGenerator`. Existing tests will be migrated as well. Authorship of the feature will be preserved through a co-authored commit.

## Impact

New

### PR DESCRIPTION
**Pull request by gambka**
Closes #26641

@jiangwen84 
@GiudGiud

### CODE REVIEW

**Comment by GiudGiud** [comment]
StitchBoundary not stitchedBoundary please. 

We try to get this action + object name in the MG now

File: framework/include/meshgenerators/StitchBoundaryMeshGenerator.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+/*                                               */
+/*    (c) 2015 Battelle Energy Alliance, LLC     */
+/*            ALL RIGHTS RESERVED                */
+/*                                               */
+/*   Prepared by Battelle Energy Alliance, LLC   */
+/*     Under Contract No. DE-AC07-05ID14517      */
+/*     With the U. S. Department of Energy       */
+/*                                               */
+/*     See COPYRIGHT for full restrictions       */
+/*************************************************/
```
Comment (GiudGiud) [inline framework/include/meshgenerators/StitchBoundaryMeshGenerator.h:14]: replace with MOOSE copyright

File: test/tests/meshgenerators/stitch_boundary_mesh_generator/tests [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      detail = "in 2D, and"
+    []
+    [stitch_3d]
+      type = 'Exodiff'
+      input = 'stitch_3d.i'
+      cli_args = '--mesh-only'
+      mesh_mode = 'REPLICATED'
+      recover = false
+      exodiff = 'stitch_3d_in.e'
+      detail = "3D."
```
Comment (GiudGiud) [inline test/tests/meshgenerators/stitch_boundary_mesh_generator/tests:22]:
```suggestion
      detail = "in 3D."
```

**Review by GiudGiud — APPROVED** [review]
thanks for the contribution

**Comment by gambka** [comment]
Related to the naming convention, we may want to consider renaming `StitchedMeshGenerator` to `StitchMeshGenerator` for the same reasons you described here.

**Comment by GiudGiud** [comment]
yeah we should. We need to do a sweep and rename all the non-verb ones

**Pull request merged**